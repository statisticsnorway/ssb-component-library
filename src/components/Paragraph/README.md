Paragraph
========

> A component for showing a paragraph
### Usage

#### HTML

```html
<p class="ssb-paragraph">Default text here</p>

<p class="ssb-paragraph negative">Negative text here</p>
```

#### React

```jsx harmony
<Paragraph>Default text here</Paragraph>

<Paragraph negative>Negative text here</Paragraph>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| negative | Optional boolean | Styling for dark theme |
