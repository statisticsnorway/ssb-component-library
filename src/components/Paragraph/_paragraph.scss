.ssb-paragraph {
  @include open-sans;
  color: $ssb-dark-6;
  font-size: 16px;
  line-height: 1.7;

  &.negative {
    color: $ssb-white;
  }
}
