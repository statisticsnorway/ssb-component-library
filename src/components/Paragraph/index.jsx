import React from 'react';
import PropTypes from 'prop-types';

const Paragraph = ({ children, className, negative }) => <p className={`ssb-paragraph${negative ? ' negative' : ''}${className ? ` ${className}` : ''}`}>{ children }</p>;

Paragraph.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	negative: PropTypes.bool,
};

export default Paragraph;
