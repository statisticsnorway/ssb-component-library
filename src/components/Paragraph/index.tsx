import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface ParagraphProps {
  children?: ReactNode
  className?: string
  negative?: boolean
}

const Paragraph: React.FC<ParagraphProps> = ({ children, className, negative }) => (
  <p className={classNames('ssb-paragraph', negative && 'negative', className)}>{children}</p>
)

export default Paragraph
