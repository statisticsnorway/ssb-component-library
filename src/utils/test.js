import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
