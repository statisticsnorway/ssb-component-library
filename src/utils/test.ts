import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
