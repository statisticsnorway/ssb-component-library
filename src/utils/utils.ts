// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
