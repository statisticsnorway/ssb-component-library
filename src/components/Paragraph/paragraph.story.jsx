import React from 'react'
import Paragraph from './index'

export default {
  title: 'Paragraph',
}

export const Default = () => (
  <div>
    <Paragraph>A paragraph is useful for wrapping text.</Paragraph>
    <Paragraph>
      A long text will have to be limited to 580px in width to make sure that it is easy to read and that the reader
      will be able to find the next line easily. If the lines are too long it is easy to loose track and it will become
      very annoying.
    </Paragraph>
  </div>
)
