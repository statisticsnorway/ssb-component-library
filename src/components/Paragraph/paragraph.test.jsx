import React from 'react'
import { render } from '../../utils/test'
import Paragraph from './index'

describe('Paragraph component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Paragraph>Paragraph</Paragraph>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Toggles classNames correctly', () => {
    const { asFragment } = render(
      <Paragraph className='mt-3' negative>
        Paragraph
      </Paragraph>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
