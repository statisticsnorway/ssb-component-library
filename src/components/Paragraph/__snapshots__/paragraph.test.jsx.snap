// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component Matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="ssb-paragraph"
  >
    Paragraph
  </p>
</DocumentFragment>
`;

exports[`Paragraph component Toggles classNames correctly 1`] = `
<DocumentFragment>
  <p
    class="ssb-paragraph negative mt-3"
  >
    Paragraph
  </p>
</DocumentFragment>
`;
